### DISCUSSION TITLE
Cannot run Uno Platform WASM app inside VS Code dev container

### DISCUSSION THREAD

**Discussion by marcinkurek** [unoplatform/uno#13808] [Q&A]
I created a new Uno WASM app using `dotnet new unoapp --name Dionysus --output ./Dionysus -preset=recommended -id=Dionysus -pub="Marcin Kurek" -tfm="net7.0" -presentation=mvux -markup=xaml -theme=material -platforms=wasm -tests=unit -tests=ui -server=true -toolkit=true -di=true -config=true -http=true -loc=true -cultures=es -cultures=fr -cultures="pt-BR" -log=serilog -nav=regions -auth=oidc -c=none -cpm=true -vscode=true -pwa=true -maui=false -skip=true`.

I'm currently trying to have it build in a debian-based dev container. `uno-check --target=wasm` reports no issues, however, the build fails at the server project with the following error:
```
/usr/share/dotnet/sdk/7.0.401/Microsoft.Common.CurrentVersion.targets(5167,5): error MSB3024: Could not copy the file "/workspaces/Dionysus/Dionysus.Server/obj/Debug/net7.0/apphost" to the destination file "bin/Debug/net7.0/Dionysus.Server", because the destination is a folder instead of a file. To copy the source file into a folder, consider using the DestinationFolder parameter instead of DestinationFiles. [/workspaces/Dionysus/Dionysus.Server/Dionysus.Server.csproj]
```

Has anyone encountered a problem like this before? I have it public under https://github.com/marcinkurek/unoapp-devcontainer-template/tree/main, anyone can try to spin it up in their VSC. I would appreciate any help.

I'm experiencing the same issue with a project generated via Visual Studio. (the build itself works fine in VS outside of the container)

**Comment by jeromelaban** [comment]
Thanks for the report! Could you [add a binlog of your build](https://platform.uno/docs/articles/uno-builds-troubleshooting.html)?

**Reply by marcinkurek** [reply]
Sure.
[msbuild.zip](https://github.com/unoplatform/uno/files/12730650/msbuild.zip)

**Reply by marcinkurek** [reply]
Looks like on Windows this works, because the build step is copying `Dionysus.Server\obj\Debug\net7.0\apphost.exe` to `Dionysus.Server\bin\Debug\net7.0\Dionysus.Server.exe`.

On Linux this fails because it's trying to do `Dionysus.Server\obj\Debug\net7.0\apphost` (file) to `Dionysus.Server\bin\Debug\net7.0\Dionysus.Server` (file), where `Dionysus.Server` already exists as a directory.

**Reply by jeromelaban** [reply]
Thanks for the update. Can you try setting `AssemblyName` to something else than `Dionysus.Server`, such as:
```xml
<PropertyGroup>
  <AssemblyName>Dionysus.AppServer</AssemblyName>
</PropertyGroup>
```

**Reply by marcinkurek** [reply]
Nope. It also creates a directory with the assembly name in `bin` and same thing happens

**Reply by marcinkurek** [reply]
The directory in question contains the following files:
`Dionysus.Server\bin\Debug\net7.0\Dionysus.Server\appsettings.Development.json`
`Dionysus.Server\bin\Debug\net7.0\Dionysus.Server\appsettings.json`
`Dionysus.Server\bin\Debug\net7.0\Dionysus.Server\Dionysus.Server.staticwebassets.runtime.json`

These files are duplicated and also exist in the parent directory with the rest of artifacts, where I think they _should_ be. They're both copied there as a part of the same build task. These files are also related to functionalities of appsettings configuration and WASM, so disabling those modules is not an option.

**Reply by marcinkurek** [reply]
I poked some more around the solution but got nothing. I really hope this doesn't end up an SDK issue 😬

**Reply by jeromelaban** [reply]
It may not be a .NET SDK issue, but this seems to be a feature that [can cause other problems](https://learn.microsoft.com/en-us/dotnet/core/install/macos-notarization-issues#native-apphost). You may try to disable it with:

```xml
<PropertyGroup>
  <UseAppHost>false</UseAppHost>
</PropertyGroup>
```

**Reply by jeromelaban** [reply]
Let's continue to track in https://github.com/unoplatform/uno/issues/13846
